src/Sarif/PerLanguageOption.cs

```diff
@@ -51,10 +51,28 @@ public PerLanguageOption(string feature, string name, Func<T> defaultValue, stri
                 throw new ArgumentException(nameof(name));
             }
 
-            this.Feature = feature;
-            this.Name = name;
-            this.DefaultValue = defaultValue;
-            this.Description = description;
+            Feature = feature;
+            Name = name;
+            DefaultValue = defaultValue;
+            Description = null;
+        }
+
+        public PerLanguageOption(string feature, string name, Func<T> defaultValue, string description)
+        {
+            if (string.IsNullOrWhiteSpace(feature))
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentException(nameof(name));
+            }
+
+            Feature = feature;
+            Name = name;
+            DefaultValue = defaultValue;
+            Description = description;
```

Comment:
This is true. That is why you should follow the usual pattern for overloads: The overload with fewer arguments should just call the overload with more arguments, passing default values for the additional arguments.

Now, in this case, the function is a constructor, so you need to use the C# syntax that allows a constructor to call another constructor in the same class:

```
public PerLanguageOption(string feature, string name, Func<T> defaultValue) :
    this(feature, name, defaultValue, null)
{
}

public PerLanguageOption(string feature, string name, Func<T> defaultValue, string description)
{
   ...
}
```